tests/Unit/Domain/CoordinateMaps/Test_Frustum.cpp

```diff
@@ -48,8 +49,35 @@ void test_suite_for_frustum(const bool with_equiangular_map) {
     test_suite_for_map(frustum_map);
   }
 }
+
+void test_frustum_fail() noexcept {
+  const std::array<std::array<double, 2>, 4> face_vertices{
+      {{{-2.0, -2.0}}, {{2.0, 2.0}}, {{-4.0, -4.0}}, {{4.0, 4.0}}}};
+  const CoordinateMaps::Frustum map(face_vertices, 2.0, 5.0,
+                                    OrientationMap<3>{});
+
+  // For the choice of params above, any point with z<=-1 should fail.
+  const std::array<double, 3> test_mapped_point1{{3.0, 3.0, -1.0}};
+  const std::array<double, 3> test_mapped_point2{{6.0, -7.0, -1.0}};
+  const std::array<double, 3> test_mapped_point3{{6.0, -7.0, -3.0}};
+
+  // Should work, even though it is outside the mapped frustum.
+  const std::array<double, 3> test_mapped_point4{{0.0, 0.0, 9.0}};
```

Comment:
Why do we want to guarantee this works?  People should not be relying on the ability to reverse-map points that are out of the range of the map.